sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json

```diff
@@ -19,59 +19,59 @@
         "x-ms-return-client-request-id": "true"
       },
       "RequestBody": {
-        "name": "azsnetyajaxiuy",
+        "name": "adpnetsearchstg",
         "type": "azureblob",
         "credentials": {
-          "connectionString": "DefaultEndpointsProtocol=https;AccountName=azsnetyajaxiuy;AccountKey=H4Q6\u002BwLogzoaDNWc1BLxgE7OkztuFjblhsRC\u002BKARuaAMt1T1xgdtOvxSG72LVYn9GH0WO65M4eHTMB6h2vQWFA==;EndpointSuffix=core.windows.net"
+          "connectionString": "DefaultEndpointsProtocol=https;AccountName=adpnetsearchstg;AccountKey=r72v9Brt0bNoOiXHv0wQ9EPNP52SlSxCxspjwfOYFTp\u002BzlMUFURgwUp0hNUHE/Vwnx8UZZfx4hh1n9i0zpIRzA==;EndpointSuffix=core.windows.net"
```

Comment:
This should be sanitized even if it's (usually) deleted by the time recordings get checked in.

Reply:
The account name and primary key must be persisted, as you can see in variables below or playback can't work. There would be no point sanitizing these variables since they have to appear below. They were deleted (and the primary key token will never be the same, even if the account name is recreated) before I ever pushed a commit with these changes.

Reply:
I know _you_ deleted these.  Somebody else won't in the future when they create resources locally.  We should have a Sanitizer strip the key out.  That's how all of the Storage tests work.

Reply:
Only if they record, in which case they have to get stored as variables as well. If they push commits with those recordings, they will have to remove them. Sanitizing these specific request/response lines won't change the fact they *must* be in variables. This is a result of putting the storage account creation in the ARM template. This stuff was created before hand, so it needs to get persisted for playback (even the resources were removed).

Reply:
We should just sanitize the `AccountKey` part of the connection string.  The rest of the values will be there to keep playback the same.  Then nobody can ever accidentally mess this up.

Reply:
Done. Right now, if the content is application/json I am allocating a `StringBuilder` each time which I did verify copies content. Since the lifetime of the test class is all methods for NUnit, this affects all methods. We could optimize it out at some point if this grows, or I have an idea for a helper struct (even fewer allocations) that will conditionally allocate a `StringBuilder` and either return it or the original string if we feel it necessary for tests when recording. I just didn't want to add it if no one was interested in this particular scenario.